src/main/java/core/basesyntax/strategy/ReturnOperationHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.Optional;
+
+public class ReturnOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        if (fruitTransaction == null) {
+            throw new IllegalArgumentException("FruitTransaction cannot be null");
+        }
+        String fruit = fruitTransaction.getFruit();
+        if (fruit == null || fruit.isEmpty()) {
+            throw new IllegalArgumentException("Fruit cannot be null or empty");
+        }
+        int amountToReturn = fruitTransaction.getQuantity();
```

Comment:
repeated logic in each handler, think about how to extract it to some helper method and where to put it to avoid code duplication